src/tiles/models/tilesManager.ts

```
@@ -0,0 +1,47 @@
+import { Logger } from '@map-colonies/js-logger';
+import { inject, injectable } from 'tsyringe';
+import { PgClient } from '../../clients/pgClient';
+import { Services } from '../../common/constants';
+import { IDBConfig } from '../../common/interfaces';
+
+export interface ITilesCountResponse {
+  tilesCount: number;
+}
+
+export interface ITilesCountRequest {
+  tilesBatchCount: number;
+  layerId: string;
+}
+
+@injectable()
+export class TilesManager {
+  public constructor(
+    @inject(Services.LOGGER) private readonly logger: Logger,
+    @inject(Services.DB_CONFIG) private readonly dbConfig: IDBConfig,
+    private readonly pgClient: PgClient
+  ) {
+    this.logger = logger;
+  }
+  public async getTilesCount(layerId: string): Promise<ITilesCountResponse> {
+    try {
+      const query = `SELECT "tilesCount" FROM "TilesCounter" WHERE "layerId"='${layerId}'`;
+      const result = await this.pgClient.execute<ITilesCountResponse>(query);
+      const data = result[0];
```

if the layer doesn't exist, what is the returned value from the pgClient? is it 0, undefined or throws an error?
In the end you should throw a NotFound error

---

fixed in e438346